src/database/database.cpp

```diff
@@ -0,0 +1,204 @@
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::RemoveAllCertsInfo(const aos::String& certType)
+{
+    if (!mSession || !mSession->isConnected()) {
+        return aos::ErrorEnum::eFailed;
+    }
+
+    try {
+
+        *mSession << "DELETE FROM certificates WHERE type = ?;", bind(certType.CStr()), now;
+
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::GetCertInfo(
+    const aos::Array<uint8_t>& issuer, const aos::Array<uint8_t>& serial, aos::iam::certhandler::CertInfo& cert)
+{
+    if (!mSession || !mSession->isConnected()) {
+        return aos::ErrorEnum::eFailed;
+    }
+
+    try {
+
+        CertInfo              result;
+        Poco::Data::Statement statement {*mSession};
+
+        statement << "SELECT * FROM certificates WHERE issuer = ? AND serial = ?;",
+            bind(Poco::Data::BLOB {issuer.Get(), issuer.Size()}), bind(Poco::Data::BLOB {serial.Get(), serial.Size()}),
+            into(result);
+
+        if (statement.execute() == 0) {
+            return aos::ErrorEnum::eNotFound;
+        }
+
+        ConvertToCertInfo(result, cert);
+
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+aos::Error Database::GetCertsInfo(const aos::String& certType, aos::Array<aos::iam::certhandler::CertInfo>& certsInfo)
+{
+    if (!mSession || !mSession->isConnected()) {
+        return aos::ErrorEnum::eFailed;
+    }
+
+    try {
+
+        std::vector<CertInfo> result;
+
+        *mSession << "SELECT * FROM certificates WHERE type = ?;", bind(certType.CStr()), into(result), now;
+
+        for (const auto& cert : result) {
+            aos::iam::certhandler::CertInfo certInfo {};
+            ConvertToCertInfo(cert, certInfo);
+            certsInfo.PushBack(certInfo);
+        }
+
+    } catch (const Poco::Exception& e) {
+        return AOS_ERROR_WRAP(aos::ErrorEnum::eFailed);
+    }
+
+    return aos::ErrorEnum::eNone;
+}
+
+Database::~Database()
+{
+    if (mSession && mSession->isConnected()) {
+        mSession->close();
+    }
+
+    Poco::Data::SQLite::Connector::unregisterConnector();
+}
+
+/***********************************************************************************************************************
+ * Private
+ **********************************************************************************************************************/
+
+void Database::CreateTables()
+{
+    *mSession << "CREATE TABLE IF NOT EXISTS certificates ("
+                 "type TEXT NOT NULL,"
+                 "issuer BLOB NOT NULL,"
+                 "serial BLOB NOT NULL,"
+                 "certURL TEXT,"
+                 "keyURL TEXT,"
+                 "notAfter TIMESTAMP,"
+                 "PRIMARY KEY (issuer, serial));",
+        now;
+}
+
+Database::CertInfo Database::ConvertToCertInfo(
```

Comment:
The names are ambiguous. Consider renaming conversion functions to something like: `ToAosCertInfo`, `FromAosCertInfo`, or something like that. Return `Database::CertInfo` through parameters.

Reply:
Done